indexer-app/src/cmr/indexer/services/index_service.clj

```diff
@@ -401,39 +411,49 @@
       (let [parsed-coll-concept (cp/parse-concept context coll-concept)]
         (index-concept context coll-concept parsed-coll-concept options)))))
 
-(defn- index-variable
-  "Index the associated variable concept of the given variable concept id."
-  [context variable-concept-id options]
-  (let [var-concept (meta-db/get-latest-concept context variable-concept-id)
-        parsed-var-concept (cp/parse-concept context var-concept)]
-    (index-concept context var-concept parsed-var-concept options)))
+(defn- concept-id->indexed-concept
+  "Given a concept id, index the concept to which it refers."
+  [context concept-id options]
+  (let [concept (meta-db/get-latest-concept context concept-id)
+        parsed-concept (cp/parse-concept context concept)]
+    (index-concept context concept parsed-concept options)))
 
 (defn- index-associated-variable
   "Index the associated variable concept of the given variable association concept."
   [context concept options]
-  (index-variable context (get-in concept [:extra-fields :variable-concept-id]) options))
+  (concept-id->indexed-concept
+   context (get-in concept [:extra-fields :variable-concept-id]) options))
+
+(defn- index-associated-service
+  "Index the associated service concept of the given service association concept."
+  [context concept options]
+  (concept-id->indexed-concept
+   context (get-in concept [:extra-fields :service-concept-id]) options))
 
 (defn- reindex-associated-variables
   "Reindex variables associated with the collection"
   [context coll-concept-id coll-revision-id]
   (let [var-associations (meta-db/get-associations-by-collection-concept-id
                           context coll-concept-id coll-revision-id :variable-association)]
     (doseq [association var-associations]
-      (index-variable context (get-in association [:extra-fields :variable-concept-id]) {}))))
+      (concept-id->indexed-concept
+       context (get-in association [:extra-fields :variable-concept-id]) {}))))
 
 (defmethod index-concept :tag-association
-  [context concept parsed-concept options]
+  [context concept _parsed-concept options]
   (index-associated-collection context concept options))
 
 (defmethod index-concept :variable-association
-  [context concept parsed-concept options]
+  [context concept _parsed-concept options]
   (index-associated-collection context concept options)
   (index-associated-variable context concept {})
   (index-associated-variable context concept {:all-revisions-index? true}))
 
 (defmethod index-concept :service-association
   [context concept parsed-concept options]
-  (index-associated-collection context concept options))
+  (index-associated-collection context concept options)
+  (index-associated-service context concept {})
```

Comment:
There is no collection related info in services index, there is no need to do this for services.

Reply:
Thanks, @ygliuvt!

Turns out this is what was also breaking that one test I couldn't figure out!